sphinxcontrib/needs/directives/needgantt.py

```diff
@@ -81,14 +81,15 @@ def run(self):
 
         timeline = self.options.get('timeline', None)
         timeline_options = ['daily', 'weekly', 'monthly']
-        if timeline is not None and timeline != '':
-            if timeline not in timeline_options:
-                raise NeedGanttException('Given scale value {} is invalid. Please use: '
-                                         '{}'.format(timeline, ",".join(timeline_options)))
+        if timeline and timeline not in timeline_options:
+            raise NeedGanttException(
+                "Given scale value {} is invalid. Please use: "
+                "{}".format(timeline, ",".join(timeline_options))
+            )
         else:
             timeline = None  # Timeline/scale not set later
```

Comment:
these are not *exactly* equivalent

the original code handles one extra case - that `timeline` is 'falsey' *and* is in `timeline_options`. But since that case can never be met, we can prune that branch